src/frontend/src/catalog/source_catalog.rs

```diff
@@ -93,7 +95,8 @@ impl From<&PbSource> for SourceCatalog {
             .into_iter()
             .map(Into::into)
             .collect();
-        let with_options = WithOptions::new(prost.properties.clone());
+        let properties = WithOptions::new(prost.properties.clone()).into_inner();
+        let options = WithOptions::new(prost.options.clone()).into_inner();
```

Comment:
BTW, should we consider backward compatibility here?

Reply:
For now, legacy codes that try to find keys of `format_encode_options` in `with_properties` are identified and corrected. So unmerging `format_encode_options` from `with_properties` will not result in key-not-found-like errors. Is this the `backward compatibility` you suggested?

Reply:
IMO the most important thing is: already created source can still work. It seems compute node doesn't use `format_encode_options` at all and thus not broken? (I'm not very sure. Correct me if I'm wrong.)

Besides is the semantics change on the frontend. I'm still not sure whether sth is changed (i.e., can work previously, but can't work now, and vice versa). But actually that's acceptable to me...

Glad to hear others' thoughts about detailed backward compatibility concerns. @BugenZhao zkss

Reply:
> already created source can still work.

Yes!

> It seems compute node doesn't use `format_encode_options` at all and thus not broken?

Also the meta service part: does it handle the cases that the new fields do not exist?

TBH, I'm not that familiar with the connector part. 🤣 Practice is the sole criterion for testing truth.

Reply:
I just added a backwards-compat test for invalid WITH options #13824. Can we or do we need to add similar stuff for this PR?

Reply:
> IMO the most important thing is: already created source can still work. 

Do you mean that, for example, a user creates a source using RW 1.4, and then upgrades to RW 1.5. So codes in RW 1.5 should recognize it as legacy source and try to find option keys in properties. Is this hot-update issue you concerned?

Reply:
Yes

Reply:
Got it. We do need to add more codes to handle this issue.

Reply:
FYI 😇 this is my solution for rejecting unknown WITH
 https://github.com/risingwavelabs/risingwave/commit/2e39b51e151b256d9007b2466d57647efabccbbd#diff-a80dd5cfb3d2b493a15182ad5c447741b10820fa16b8eea2719b9c0ae6f003a4R415-R418